### PR TITLE
Goth examples on CI

### PR DESCRIPTION
**Pull request by approxit**
What I've done:
- Added all non-outbound vm examples on CI to be run nightly at 2 AM
- Removed default usage of outbound pypi URL (to ease example running)
- Fixed https://github.com/golemfactory/ray-on-golem/issues/197
- Made SSH check fails able to retry activity creation
- Removed hostname resolution dependency in `simple-task` and `simple-actor` examples, as sometimes goth instances had problems with their stability
- Fixed outdated `node_config` layout in mendelbrot yaml example
- Made `node_config` raise errors on unrecognized fields (to mitigate the previous element from this list)
- Fixed CI warnings about outdated node versions, by bumping versions of a few GHA steps
- Made yagna running checks explicit about the used URL
- Fixed problems with failing yagna funds on CI, by making its timeout a little bigger

Notable remarks:
- We decided that examples will be run not on pull requests and as nighty due to current problems with ssh stability

### CODE REVIEW

File: golem-cluster.yaml [hunk 1]
```diff
@@ -35,7 +35,7 @@ provider:
         # List of urls which will be added to the Computation Manifest
         # Requires protocol to be defined in all URLs
         # If not provided demand will not use Computation Manifest
-        outbound_urls: ["https://pypi.dev.golem.network"]
+        # outbound_urls: ["https://pypi.dev.golem.network"]
```
Comment (lucekdudek) [inline golem-cluster.yaml:38]: Why this change? Don't we use overwrite anyway?

File: golem-cluster.override.2-image.yaml [hunk 2]
```diff
@@ -2,4 +2,4 @@ provider:
   parameters:
     node_config:
       demand:
-        image_tag: "blueshade/ray-on-golem:0.8.0-py3.10.13-ray2.9.2"
+        image_tag: "golem/ray-on-golem:0.8.0-py3.10.13-ray2.9.2"
```
Comment (lucekdudek) [inline golem-cluster.override.2-image.yaml:5]: I think this is important for pre release versions to be not from the `golem` user registery.

File: pyproject.toml [hunk 3]
```diff
@@ -9,9 +9,9 @@ packages = [{include = "ray_on_golem"}]
 [tool.poetry.dependencies]
 python = "^3.8.1"
 
-ray = {version="^2.9.2", extras=["default"]}
-#golem-core = {path="../golem-core-python", develop=true}
-#golem-core = {git="https://github.com/golemfactory/golem-core-python.git", branch="main"}
+ray = {version = "^2.9.2", extras = ["default"]}
+#golem-core = {path = "../golem-core-python", develop = true}
+#golem-core = {git = "https://github.com/golemfactory/golem-core-python.git", branch = "main"}
```
Comment (lucekdudek) [inline pyproject.toml:14]: My formatter would also change `#golem-core` into `# golem-core`. What are you using?

File: pyproject.toml [hunk 4]
```diff
@@ -56,6 +57,11 @@ dpath = "^2.1.6"
 yamlpath = "^3.8.1"
 pytest = "^7.4.3"
 
+[[tool.poetry.source]]
+name = "test"
```
Comment (lucekdudek) [inline pyproject.toml:61]: Maybe `test-pypi` ?

File: .github/workflows/start-goth.sh [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+rm -rf .envs/goth
+python -m venv .envs/goth
+source .envs/goth/bin/activate
+
+echo INSTALLING TOOLS
+python -m pip install --upgrade pip
+python -m pip install --upgrade setuptools wheel
+
+echo INSTALLING DEPENDENCIES
+python -m pip install goth==0.15.11 pytest pytest-asyncio pexpect
```
Comment (lucekdudek) [inline .github/workflows/start-goth.sh:24]: Maybe we can move the goth version to params so it is easier to upgrade (or even run on multiple versions) later?

File: .github/workflows/integration_tests.yml [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+            args: 2c26b46b68ffc68ff99b453c1d30413413422d706483bfa0f98a5e886266e7ae
+
+          - example_name: mandelbrot_image
+            base_yaml_path: examples/mandelbrot-golem-cluster.yaml
+
+#          - example_name: outbound
+#            base_yaml_path: examples/outbound.yaml
+#
+#          - example_name: outbound_tester
+#            base_yaml_path: examples/outbound_tester.yaml
```
Comment (lucekdudek) [inline .github/workflows/integration_tests.yml:29]: Add a reason comment why those are disabled pls

File: .github/workflows/integration_tests.yml [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+
+      - name: Collects logs
+        if: always()
+        uses: actions/upload-artifact@v4
+        with:
+          name: logs-example-${{ matrix.example_name }}
+          path: |
+            /root/.local/share/ray_on_golem/webserver_debug.log
+            /root/.local/share/ray_on_golem/yagna.log
+            /tmp/goth-tests
```
Comment (lucekdudek) [inline .github/workflows/integration_tests.yml:92]: Shouldn't we stop goth after all of this?

File: pyproject.toml (hunk 3 above)
Comment (approxit) [reply pyproject.toml:14]: Well, PyCharm does not use space there. It would be great if we could use autoformat here...

File: .github/workflows/integration_tests.yml (hunk 7 above)
Comment (approxit) [reply .github/workflows/integration_tests.yml:92]: Do you have any example for stopping goth?

Comment (lucekdudek) [reply .github/workflows/integration_tests.yml:92]:
```yaml
    - name: Remove Docker containers
      shell: bash
      if: always()
      continue-on-error: true
      run: docker rm --force $(docker ps -a -q)

    - name: Stop goth network
      shell: bash
      if: always()
      continue-on-error: true
      run: kill $(ps -aux | grep "[g]oth start" | awk '{print $2}')
```

File: .github/workflows/integration_tests.yml [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+          - example_name: simple-actor
+
+          - example_name: simple-task
+
+          # Commented out as Goth does not support outbound yet
+          # - example_name: dds-with-ray
+          #  - example_name: outbound
+          #    base_yaml_path: examples/outbound.yaml
+          #  - example_name: outbound_tester
+          #    base_yaml_path: examples/outbound_tester.yaml
```
Comment (lucekdudek) [inline .github/workflows/integration_tests.yml:33]: Formatting looks off
```python
# - example_
... vs
#  - example_
```

File: .github/workflows/integration_tests.yml [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+
+jobs:
+  examples:
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - example_name: calculate_pi
+
+
```
Comment (lucekdudek) [inline .github/workflows/integration_tests.yml:17]: Extra empty line

File: .github/workflows/stop-goth.sh [hunk 10]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+echo STOP GOTH
+kill $(ps -aux | grep "[g]oth start" | awk '{print $2}')
+
+echo STOP DOCKER CONTAINERS
+docker rm --force $(docker ps -a -q)
```
Comment (lucekdudek) [inline .github/workflows/stop-goth.sh:6]: I wonder if we can get PID from this
```
python -m goth start .envs/goth/assets/goth-config.yml &
GOTH_PID=$!
```

Or if there is a symetric command to the `goth start`

File: pyproject.toml [hunk 11]
```diff
@@ -23,15 +23,16 @@ appdirs = "^1.4.4"
 
 websocat = "^1.12.0"
 golem-node = "0.14.0"
+#golem-node = {version = "0.15.0a18", source = "test"}
```
Comment (lucekdudek) [inline pyproject.toml:26]: source = "test-pipy"

**Review by lucekdudek — APPROVED** [review]
